RebornCore/src/main/java/reborncore/common/blockentity/SlotConfiguration.java

```diff
@@ -46,10 +51,16 @@
 public class SlotConfiguration implements NBTSerializable {
 
 	List<SlotConfigHolder> slotDetails = new ArrayList<>();
-
+	private static final HashMap<Map.Entry<ServerWorld, BlockPos>, BlockApiCache<Storage<ItemVariant>, Direction>> sidedStorageCache = new HashMap<>();
```

Comment:
Please don't do this, this is a memory leak.

Reply:
I think what would work nicely is ensuring that the inventories are only queried once per tick, e.g. by storing them in a `Storage<ItemVariant>[6]`.

Reply:
I guess its better to have it inside machineBlockEntity right? But still sadly there's a chance that Storage itself can be removed or changed so caching BlockApiCache instead...

Reply:
There is no need to cache it if it's not practical. I'm saying that instead of it being queried for each slot, it could be queried once for the whole configuration.

Reply:
I think you forgot to commit some changes in your last commit, but it's looking good. When I wrote BlockApiCache, I intended it to be stored in the machine BE.

Reply:
I once tried caching Storage itself and updated it when its neighbor changed but some weirdness still existed... especially Create mod (well)

Reply:
That would be incorrect indeed. The storage is only valid for a short (unspecified) amount of time, and it can change without neighbor updates.